core/shared/platform/include/platform_wasi.h

```diff
@@ -1093,6 +1093,32 @@ os_is_handle_valid(os_file_handle *handle);
 char *
 os_realpath(const char *path, char *resolved_path);
 
+/****************************************************
+ *                                                  *
+ *                Clock functions                   *
+ *                                                  *
+ ****************************************************/
+
+/**
+ * Get the resolution of the specified clock.
+ *
+ * @param clock_id clock identifier
+ * @param resolution output variable to store the clock resolution
+ */
+__wasi_errno_t
+os_clock_res_get(__wasi_clockid_t clock_id, __wasi_timestamp_t *resolution);
+
+/**
+ * Get the current time of the specified clock.
+ *
+ * @param clock_id clock identifier
+ * @param resolution clock identifier
```

Comment:
Should we have here the documentation for the precision param?

Reply:
added documentation

Reply:
But the resolution then should be deleted, right?

Reply:
yes thanks for spotting - have removed resolution 👍